Plant Expansion Beta: V.02

Spankmaster1066 opened the pull request:
[Changelogs]: Hi everyone I forgot to write a changelog for the first installment of the Plant Expansion (part of the Lean Extended Universe). Here's the basics:
1. Potted plants can now be emagged, this allows you to hold them with one hand. 
2. You can now screwdrive open a potted plant and insert a sharp item into it to create a sharp bush-weapon. 
3. Feeding an emagged potted plant lean will bring it to life. 

:cl: Spankmaster
tweak: Potted Lean behavior, adds chance to inject a variable amount of lean, tweaks some of the speech verbs and stuff
fix: Fixed a bug where you could infinitely sharpen a potted plant. 
tweak: Potted Plants now inherit the force of the item inserted into them. Fixed some of the outputted messages as well for performing actions on plants.
/:cl:

[why]: # Makes the experience of working with potted plants more fun and realistic.

JohnGinnane commented on hippiestation/code/modules/mob/living/simple_animal/hostile/retaliate/potted_lean.dm:
```diff
@@ -5,22 +5,34 @@
 	attack_sound = 'hippiestation/sound/effects/plantAttack.ogg'
 	butcher_results = list(/obj/item/reagent_containers/food/snacks/grown/grapes = 3)
 	gold_core_spawnable = HOSTILE_SPAWN
+	var/attackInject = "lean"
```
Kinda nitpicky but could this be changed to `attack_inject` to be consistent with the rest of the vars?

Spankmaster1066 replied:
Sure thing, I am still learning the little standards we use on here.

JohnGinnane commented:
Code is 🆗 but will wait for any other changes before approving because it'll be dismissed anyway

Spankmaster1066 commented:
Alright alright CHNKR I will set that up tonight, this still is kinda seperate. It just fixes the functionality we have on the server currently but alright gimme a sec

Chnkr commented:
@Spankmaster1066 Thanks dude, because when you do that you'll fix the issue with picking it up as well 👍

Spankmaster1066 commented:
I hear ya bro, the only thing is I have a fuckton of real life android coding assignments to work on right now so I can't go in and tweak this. So I will write it in a day or two and merge it to this.

That aside though, if this is merged now it will solve all the issues for the currently implemented potted plants besides the one-handed thing. They're a tad broken right now but it isn't game-breaking.

Spankmaster1066 commented:
OK Chnkr your request is met and this PR is 100% fully functional, no bugs at all as far as I can tell.

Chnkr commented on hippiestation/code/game/objects/items/dildo.dm:
```diff
@@ -1,48 +1 @@
-//Coded by Nexendia~ :)
```
No.

Chnkr commented on hippiestation/code/game/objects/structures/flora.dm:
```diff
@@ -61,4 +60,88 @@
+	var/is_sharpened = 0 //to prevent multiple sharpenings
+	var/sharp_prefix = "thorny"
+	
+obj/item/kirbyplants_onehanded/equipped(mob/living/user)
```
Missing `/`

Chnkr commented on hippiestation/code/game/objects/structures/flora.dm:
```diff
@@ -61,4 +60,88 @@
+	I.layer = ABOVE_MOB_LAYER
+	..()
+
+obj/item/kirbyplants_onehanded/dropped(mob/living/user)
```
Missing `/`

Chnkr commented on hippiestation/code/game/objects/structures/flora.dm:
```diff
@@ -61,4 +60,88 @@
+	..()
+	user.remove_alt_appearance("sneaking_mission")
+	
+obj/item/kirbyplants_onehanded/random
```
Missing `/` and the `random` path is not needed, since these will never be dropped as a random plant

Chnkr commented on hippiestation/code/game/objects/structures/flora.dm:
```diff
@@ -61,4 +60,88 @@
+	icon_state = "random_plant"
+	var/list/static/states
+	
+obj/item/kirbyplants_onehanded/random/Initialize()
```
Missing `/` and the `random` path is not needed, since these will never be dropped as a random plant

Chnkr commented on hippiestation/code/game/objects/structures/flora.dm:
```diff
@@ -61,4 +60,88 @@
+		generate_states()
+	icon_state = pick(states)
+
+obj/item/kirbyplants_onehanded/random/proc/generate_states()
```
Missing `/` and the `random` path is not needed, since these will never be dropped as a random plant

bardofrage commented:
add this fix to our great station +1

JohnMcLaime commented:
This makes chameleon device pointless, with emag you basically get its a bit weaker (to thermals/sechud) version.

fluxcapacitor1337 commented:
+1

McBawbaggings commented:
we need 2 sell this as DLC under hippiecoin cryptocurrency
+1

JohnGinnane commented:
@McBawbaggings does this even need votes if it's bugfixing on already-approved stuff?

Pyko1 commented:
Undocumented code changes, this removes code that is not listed on initial pr or changes.

McBawbaggings commented:
HOLY SHIT I DIDN'T EVEN NOTICE LMAO

spankmonkey ur gonna meet me behind mcdonalds and we're gonna fight 2 the death now

JohnGinnane commented:
Whoops, didn't even realise other commits were done

Spankmaster1066 commented:
I deleted Dildo's on my own Repo and forgot it would transfer over here. Let me fix that. I'm sorry I deleted dildos everyone.


To Roma, what you say is half true. The Chameleon Projector has the utility of being able to assume the form of ANY item, whereas this only lets you hide in a bush. Anyone who is aware of you using this strategy will instantly key in on bushes. a chameleon projector allows you to do plays like shooting someone then bolting into maint and transforming into a TINY and boring object, like a wire clipping.

Spankmaster1066 commented:
OK, all checks passed, Dildo stuff is fixed. Admins let's do this.

banthebantz commented:
+1
